### DISCUSSION TITLE
Problem connecting to slingbox not on same network

### DISCUSSION THREAD

**Discussion by hanchan07** [GerryDazoo/Slinger#35] [Q&A]
My brother-in-law has a slingbox M1 in Japan that we have been connecting to to watch live Japanese TV and DVR in America.  We were happy to see this solution so we can continue to do this however I am not able to connect to the slingbox.

I am making a few assumptions here as I have not seen an example of what I am trying to do.

- I am using the admin password for the slingbox not the user password.
- I uncommented line 9 and entered the public address of the slingbox since it doesn't make sense to use a local since I am not on the same network
- I uncommented line 10 and entered the port number that is noted in the network info that is displayed in the slingbox software we currently use to watch. The port displayed is 5301.
- I uncommented line 12 for the finder id since it was commented out as is and entered the finderId that I got from the link in the instructions.
- I uncommented line 14 for the name and entered the name of the slingbox as it appeared in the sling software, I assume this doesnt matter what is put here anyway.

After making the changes to the config.ini and running the .exe it appears that the finderId is ok, but gives the error:
`Error connecting to (xx.xx.xx.xx, 5301)`
I am excluding the actual IP here for obvious reasons.
Other errors then follow since it is not able to establish a connection.

From what I have read this should be possible, or does my Brother-in-law need to run the server on his network and I connect to that IP through VLC?

After reading the v3.0 Release Notes PDF it seems like my bro-in-law is the one that needs to run the server.  Can someone confirm or give any suggestions?

Thanks in advance.

**Comment by egadget1** [comment]
Hi, 
   I think you can leave lines 9 and 10 commented out. Run the server on the computer with the slingbox on the same internal network.  You then need to log into your router and enable port forwarding.  Type in the computer IP (ie local 192.168.1.xx) and 8080 to the port forwarding fields.  Then you can open up VLC and select open network stream and type in your public address; ie http://74.97.39.xxx:8080/slingbox, ie replace the 74.97.39.xxx with your ipaddress  (in goggle just type in whats my ip, and it will tell you your public ip address).  Hope this helps.

**Reply by hanchan07** [reply]
Yeah I think that's probably what I will have to do, I was hoping to run the server over here since I don't think my brother-in-law wants to leave his computer on all day.  But maybe we can get him to do it until the replacement device comes back in stock.

I may have him try enabling port forwarding with the port the slingbox is on and see if the server will connect that way.  It's worth a shot.

Thanks.

**Reply by GerryDazoo** [reply]
For sure that will work. Many people including me do it.

**Reply by hanchan07** [reply]
@GerryDazoo 
Ok I got the server running on my desktop to connect to his IP.  Since port 5301(the one sling is using) was not available we used 4816 and forwarded that to the internal port 5301, then added a rule in the router to allow access from my IP to port 4816.
The server now is able to connect to his IP xx.xx.xx.xx:4816 with the message:
`* Running on http://192.168.86.32:8081`
However when I connect to the sling server via VLC using `http://192.168.86.32:8080/slingbox` I get the following error:

```javascript
HideKatze box Starting Stream for 192.168.86.32:55480
Connecting... ('xx.xx.xx.xx', 4816) control
cmd: 0x67 err: 0x4f20 21258
cmd: 0xc6 err: 0x742f 12092
Hidekatze box Error Starting Session. Check your admin password in config.ini file!
Hidekatze box ERROR: Slingbox session startup failed. 
```

I have double checked that I had the admin password correct and I do have the admin password entered in the config file as was in output using the provided url to get the credentials for the slingbox.

**Reply by GerryDazoo** [reply]
I suspect the 5301 port forwarding rule was there already pointing to your brothers slingbox.

From the errors it looks like it failing on the very first command to the slingbox and then the password fails. I don't think you are  connecting to the slingbox you think you are. Or that it's even a slingbox.
Start a video session to it with the sling software and check the setting/network info. You may be surprised. 
Just for grins, change the slingbox port number in your config.ini back to 5301 and give that a try.

**Reply by hanchan07** [reply]
@GerryDazoo I had him add the port forwarding rule, removing it and changing the config file back to 5301 resulted in not being connect to his router at all. After spending the last few weeks looking into the issue we discovered that the router listed available ports but all of the 5000 ports were not in that list.  Upon further research he has IPv6 enabled on his router and with IPv6 certain ranges of ports become unavailable.  Unfortunately his ISP requires IPv6 so we can't turn it off.  It appears that IPv6 providers are common in Asia.  
At this point our only option is to have him run the server on his network.

Thanks for the help

**Comment by GerryDazoo** [comment]
I’m very surprised about this. Yes, most routers have a list of commonly used ports and services to make it easier to setup up things like SSH or FTP. But every router I have every come across also allows the user to setup up their own custom service with a particular port in the range 1024-65535.

Please connect to the remote slingbox with the Sling supplied software and report the box's "Network Info"

*	Gerry

**Reply by GerryDazoo** [reply]
I will look into adding IPV6 support for the slingbox connection. I'm still trying to figure out how I could test it here locally. Turns out my ISP doesn't do IPV6.

**Reply by GerryDazoo** [reply]
On widows you can set up a IV4 to IPV6 proxy connection to get your slingbox packets to the IPV6 gateway

https://learn.microsoft.com/en-us/windows-server/networking/technologies/netsh/netsh-interface-portproxy

**Comment by GerryDazoo** [comment]
Your brother need to add a port forwarding rule on his router to map port 5301 to the internal IP address of the slingbox.

Your brother doesn’t need to run the server. Many people have this working with a local server and a remote slingbox. 

 

 

From: Tyler Hansen ***@***.***> 
Sent: Sunday, September 04, 2022 2:41 AM
To: GerryDazoo/Slinger ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [GerryDazoo/Slinger] Problem connecting to slingbox not on same network (Discussion #35)

 

My brother-in-law has a slingbox M1 in Japan that we have been connecting to to watch live Japanese TV and DVR in America. We were happy to see this solution so we can continue to do this however I am not able to connect to the slingbox.

I am making a few assumptions here as I have not seen an example of what I am trying to do.

*	I am using the admin password for the slingbox not the user password.
*	I uncommented line 9 and entered the public address of the slingbox since it doesn't make sense to use a local since I am not on the same network
*	I uncommented line 10 and entered the port number that is noted in the network info that is displayed in the slingbox software we currently use to watch. The port displayed is 5301.
*	I uncommented line 12 for the finder id since it was commented out as is and entered the finderId that I got from the link in the instructions.
*	I uncommented line 14 for the name and entered the name of the slingbox as it appeared in the sling software, I assume this doesnt matter what is put here anyway.

After making the changes to the config.ini and running the .exe it appears that the finderId is ok, but gives the error:
Error connecting to (xx.xx.xx.xx, 5301)
I am excluding the actual IP here for obvious reasons.
Other errors then follow since it is not able to establish a connection.

From what I have read this should be possible, or does my Brother-in-law need to run the server on his network and I connect to that IP through VLC?

After reading the v3.0 Release Notes PDF it seems like my bro-in-law is the one that needs to run the server. Can someone confirm or give any suggestions?

Thanks in advance.

**Comment by GerryDazoo** [comment]
Assuming he forwards the port correctly it will work. Done it many times.

 

*	Gerry
